<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the Ontology Lookup Service â€¢ rols</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An R interface to the Ontology Lookup Service">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rols</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">2.9.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rols.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="rols_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="rols_files/jquery-1.12.4/jquery.min.js"></script><link href="rols_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="rols_files/datatables-binding-0.4/datatables.js"></script><link href="rols_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rols_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rols_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="rols_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="rols_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>An R interface to the Ontology Lookup Service</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      <a href="https://lgatto.github.io/about/" class="uri">https://lgatto.github.io/about/</a><br><div class="hidden name"><code>rols.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query the Ontology Lookup Service directly from R and how to create and parse controlled vocabulary.</p>
    </div>
    
<pre><code>## Warning: package 'GO.db' was built under R version 3.6.0</code></pre>
<pre><code>## Warning: package 'DT' was built under R version 3.6.0</code></pre>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> is a Bioconductor package and should hence be installed using the dedicated functionality</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## try http:// if https:// URLs are not supported
<span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))
    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"rols"</span>)</code></pre></div>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-help" class="anchor"></a>Getting help</h2>
<p>To get help, either post your question on the <a href="https://support.bioconductor.org/">Bioconductor support site</a> or open an issue on the <em><a href="http://bioconductor.org/packages/rols">rols</a></em> <a href="https://github.com/lgatto/rols/issues">github page</a>.</p>
</div>
<div id="the-resource" class="section level2">
<h2 class="hasAnchor">
<a href="#the-resource" class="anchor"></a>The resource</h2>
<p>The <a href="http://www.ebi.ac.uk/ontology-lookup/">Ontology Lookup Service</a> (OLS) [<a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-97">1</a>, <a href="http://nar.oxfordjournals.org/content/36/suppl_2/W372">2</a>] is originally spin-off of the <a href="http://www.ebi.ac.uk/pride/archive/">PRoteomics IDEntifications</a> database (PRIDE) service, located at the EBI, and is now developed and maintained by the <a href="http://www.ebi.ac.uk/about/spot-team">Samples, Phenotypes and Ontologies</a> team at EMBL-EBI.</p>
</div>
<div id="the-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-package" class="anchor"></a>The package</h2>
<p>The OLS provides a REST interface to hundreds of ontologies from a single location with a unified output format. The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package make this possible from within R. Do do so, it relies on the <em><a href="https://CRAN.R-project.org/package=httr">httr</a></em> package to query the REST interface, and access and retrieve data.</p>
<p>There are 214 ontologies available in the OLS, listed in the table below. Their name is to be use to defined which ontology to query.</p>
<div id="htmlwidget-4a9590b472cccbae55f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4a9590b472cccbae55f5">{"x":{"filter":"none","data":[["aero","bfo","co_330","co_334","co_335","co_360","ddanat","dideo","dinto","eco","gaz","genepio","geno","hom","mfomd","mi","micro","mondo","mop","mp","mpath","ms","oae","oarcs","oba","obi","obib","ogsf","ohmi","omp","sbo","scdo","sepio","vo","vt","vto","xao","agro","ero","fbbt","ancestro","fbdv","apo","aro","fypo","idomal","ino","bao","kisao","bco","ma","cdao","mfmo","cheminf","mfoem","cio","cmo","mirnao","co_320","miro","co_321","co_322","mmusdv","co_323","mod","co_333","co_336","co_347","mro","co_356","ncbitaxon","co_359","ncit","cvdo","omiabis","ddpheno","doid","aeo","duo","ecocore","emap","atol","eo","eupath","bcgo","bspo","bto","caro","omit","cco","ceph","omrse","chebi","fbbi","chmo","ontoneo","gexo","oostt","go","iao","opl","ordo","mf","cl","mmo","ovae","clo","mpio","cmpo","pco","nbo","co_324","peco","obcs","po","co_325","ogg","co_327","poro","ogi","co_331","so","co_337","ogms","co_338","spd","ohd","co_339","co_340","tads","co_341","co_343","olatdv","co_345","trans","pato","co_346","unimod","co_348","plana","co_350","ppo","wbbt","pr","probonto","wbls","co_357","pw","cro","zfa","reto","cteno","dicom","rex","zfs","rexo","dpo","rnao","dron","ro","edam","rs","rxno","efo","sep","ehdaa2","sibo","emapa","enm","swo","ensemblglossary","envo","uberon","eol","epo","exo","xl","fao","fbcv","fix","flopo","flu","fma","foodon","geo","hancestro","hao","hp","hsapdv","ico","ido","lbo","mamo","miapa","ncro","nmrcv","pdro","pdumdv","pride","sdgio","sio","stato","symp","taxrank","teddy","tgma","to","tto","uo","upheno","vario","wbphenotype","xco","zeco"],["AERO","BFO","CO_330","CO_334","CO_335","CO_360","DDANAT","DIDEO","DINTO","ECO","GAZ","GENEPIO","GENO","HOM","MFOMD","MI","MICRO","MONDO","MOP","MP","MPATH","MS","OAE","OARCS","OBA","OBI","OBIB","OGSF","OHMI","OMP","SBO","SCDO","SEPIO","VO","VT","VTO","XAO","AGRO","ERO","FBbt","ancestro","FBdv","APO","ARO","FYPO","IDOMAL","INO","BAO","KISAO","BCO","MA","CDAO","MFMO","CHEMINF","MFOEM","CIO","CMO","MIRNAO","CO_320","MIRO","CO_321","CO_322","MMUSDV","CO_323","MOD","CO_333","CO_336","CO_347","MRO","CO_356","NCBITAXON","CO_359","NCIT","CVDO","OMIABIS","DDPHENO","DOID","AEO","DUO","ECOCORE","EMAP","ATOL","EO","EUPATH","BCGO","BSPO","BTO","CARO","OMIT","CCO","CEPH","OMRSE","CHEBI","FBbi","CHMO","ONTONEO","GeXO","OOSTT","GO","IAO","OPL","ORDO","MF","CL","MMO","OVAE","CLO","MPIO","CMPO","PCO","NBO","CO_324","PECO","OBCS","PO","CO_325","OGG","CO_327","PORO","OGI","CO_331","SO","CO_337","OGMS","CO_338","SPD","OHD","CO_339","CO_340","TADS","CO_341","CO_343","OLATDV","CO_345","TRANS","PATO","CO_346","UNIMOD","CO_348","PLANA","CO_350","PPO","WBBT","PR","PROBONTO","WBLS","CO_357","PW","CRO","ZFA","ReTO","CTENO","DICOM","REX","ZFS","ReXO","FBcv","RNAO","DRON","RO","EDAM","RS","RXNO","EFO","SEP","EHDAA2","SIBO","EMAPA","ENM","SWO","ENSGLOSS","ENVO","UBERON","EOL","EPO","EXO","XL","FAO","FBcv","FIX","FLOPO","FLU","FMA","FOODON","GEO","HANCESTRO","HAO","HP","HSAPDV","ICO","IDO","LBO","MAMO","MIAPA","NCRO","NMR","PDRO","PDUMDV","PRIDE","SDGIO","SIO","STATO","SYMP","TAXRANK","TEDDY","TGMA","TO","TTO","UO","UPHENO","VARIO","WBPhenotype","XCO","ZECO"],["aero","bfo","co_330","co_334","co_335","co_360","ddanat","dideo","dinto","eco","gaz","genepio","geno","hom","mfomd","mi","micro","mondo","mop","mp","mpath","ms","oae","oarcs","oba","obi","obib","ogsf","ohmi","omp","sbo","scdo","sepio","vo","vt","vto","xao","agro","ero","fbbt","ancestro","fbdv","apo","aro","fypo","idomal","ino","bao","kisao","bco","ma","cdao","mfmo","cheminf","mfoem","cio","cmo","mirnao","co_320","miro","co_321","co_322","mmusdv","co_323","mod","co_333","co_336","co_347","mro","co_356","ncbitaxon","co_359","ncit","cvdo","omiabis","ddpheno","doid","aeo","duo","ecocore","emap","atol","eo","eupath","bcgo","bspo","bto","caro","omit","cco","ceph","omrse","chebi","fbbi","chmo","ontoneo","gexo","oostt","go","iao","opl","ordo","mf","cl","mmo","ovae","clo","mpio","cmpo","pco","nbo","co_324","peco","obcs","po","co_325","ogg","co_327","poro","ogi","co_331","so","co_337","ogms","co_338","spd","ohd","co_339","co_340","tads","co_341","co_343","olatdv","co_345","trans","pato","co_346","unimod","co_348","plana","co_350","ppo","wbbt","pr","probonto","wbls","co_357","pw","cro","zfa","reto","cteno","dicom","rex","zfs","rexo","dpo","rnao","dron","ro","edam","rs","rxno","efo","sep","ehdaa2","sibo","emapa","enm","swo","ensemblglossary","envo","uberon","eol","epo","exo","xl","fao","fbcv","fix","flopo","flu","fma","foodon","geo","hancestro","hao","hp","hsapdv","ico","ido","lbo","mamo","miapa","ncro","nmrcv","pdro","pdumdv","pride","sdgio","sio","stato","symp","taxrank","teddy","tgma","to","tto","uo","upheno","vario","wbphenotype","xco","zeco"],["Adverse Event Reporting Ontology","Basic Formal Ontology","Potato ontology","Cassava ontology","Common bean ontology","Sugar Kelp trait ontology","Dictyostelium discoideum anatomy","Drug-drug Interaction and Drug-drug Interaction Evidence Ontology","The Drug-Drug Interactions Ontology","Evidence &amp; Conclusion Ontology (ECO)","Gazetteer","Genomic Epidemiology Ontology","Genotype Ontology","Homology Ontology","Mental Disease Ontology","Molecular Interactions Controlled Vocabulary","Ontology of Prokaryotic Phenotypic and Metabolic Characters","MONDO: Monarch Disease Ontology","Molecular Process Ontology","The Mammalian Phenotype Ontology","Mouse pathology ontology","Mass spectrometry ontology","OAE: Ontology of Adverse Events","Ontology of Arthropod Circulatory Systems","Ontology of Biological Attributes","Ontology for Biomedical Investigations","Ontology for BIoBanking (OBIB)","Ontology of Genetic Susceptibility Factors (OGSF)","OHMI: Ontology of Host-Microbiome Interactions","Ontology of Microbial Phenotypes","Systems Biology Ontology","Sickle Cell Disease Ontology","Scientific Evidence and Provenance Information Ontology","Vaccine Ontology","Vertebrate trait ontology","Vertebrate Taxonomy Ontology","Xenopus Anatomy Ontology","Agronomy Ontology","eagle-i resource ontology","Drosophila gross anatomy","Ancestry Ontology","Drosophila development","Ascomycete phenotype ontology","Antibiotic Resistance Ontology","Fission Yeast Phenotype Ontology","Malaria Ontology","INO: Interaction Network Ontology","BioAssay Ontology","Kinetic Simulation Algorithm Ontology","Biological Collections Ontology","Mouse adult gross anatomy","Comparative Data Analysis Ontology","Mammalian Feeding Muscle Ontology","chemical information ontology (cheminf) - information entities about chemical entities","Emotion Ontology","Confidence Information Ontology","Clinical measurement ontology","microRNA Ontology","Rice ontology","Mosquito insecticide resistance","Wheat ontology","Maize ontology","Mouse Developmental Stages","Barley ontology","Protein modification","Beet Ontology ontology","Soybean ontology","Castor bean ontology","MHC Restriction Ontology","Vitis ontology","NCBI organismal classification","Sunflower ontology","NCI Thesaurus OBO Edition","Cardiovascular Disease Ontology","Ontologized MIABIS","Dictyostelium discoideum phenotype ontology","Human Disease Ontology","Anatomical Entity Ontology","The Data Use Ontology","An ontology of core ecological entities","Mouse gross anatomy and development, timed","Animal Trait Ontology for Livestock","Plant Environment Ontology","EuPathDB ontology","Beta Cell Genomics Ontology","Biological Spatial Ontology","BRENDA tissue / enzyme source","Common Anatomy Reference Ontology","Ontology for MIRNA Target","Cell Cycle Ontology","Cephalopod Ontology","Ontology of Medically Related Social Entities","Chemical Entities of Biological Interest","Biological Imaging Methods Ontology","Chemical Methods Ontology","Obstetric and Neonatal Ontology","Gene Expression Ontology","Ontology of Organizational Structures of Trauma centers and Trauma Systems","Gene Ontology","Information Artifact Ontology","Ontology for Parasite LifeCycle","Orphanet Rare Disease Ontolog","Mental Functioning Ontology","Cell Ontology","Measurement method ontology","OVAE: Ontology of Vaccine Adverse Events","CLO: Cell Line Ontology","Minimum PDDI Information Ontology","Cellular Microscopy Phenotype Ontology","Population and Community Ontology","Neuro Behavior Ontology","Sorghum ontology","Plant Experimental Conditions Ontology","Ontology of Biological and Clinical Statistics","Plant Ontology","Banana ontology","OGG: Ontology of Genes and Genomes","Pearl millet ontology","Porifera (sponge) ontology","Ontology for genetic interval","Sweet Potato ontology","Sequence types and features","Groundnut ontology","Ontology for General Medical Science","Chickpea ontology","Spider Ontology","The Oral Health and Disease Ontology","Lentil ontology","Cowpea ontology","Tick Anatomy Ontology","Pigeonpea ontology","Yam ontology","Medaka Developmental Stages","Brachiaria ontology","Pathogen Transmission Ontology","Phenotype And Trait Ontology","Mungbean ontology","Unimod protein modification database for mass spectrometry","Brassica ontology","planaria-ontology","Oat ontology","Plant Phenology Ontology","C. elegans Gross Anatomy Ontology","PRotein Ontology (PRO)","Probability Distribution Ontology","C. elegans development ontology","Woody Plant Ontology ontology","Pathway ontology","Contributor Role Ontology","Zebrafish anatomy and development ontology","Regulation of Transcription Ontology","Ctenophore Ontology","DICOM Controlled Terminology","Physico-chemical process","Zebrafish developmental stages ontology","Regulation of Gene Expression Ontology","Drosophila Phenotype Ontology","The RNA Ontology (RNAO): An Ontology for Integrating RNA Sequence and Structure Data","The Drug Ontology","OBO Relations Ontology","Bioinformatics operations, data types, formats, identifiers and topics","Rat Strain Ontology","Name Reaction Ontology","Experimental Factor Ontology","Sample processing and separation techniques","Human developmental anatomy, abstract","Social Insect Behavior Ontology","Mouse gross anatomy and development, timed","eNanoMapper Ontology","Bioinformatics operations, data types, formats, identifiers and topics","Ensembl Glossary","Environment Ontology","Uber-anatomy ontology","Environment Ontology for Livestock","Epidemiology Ontology","Exposure ontology","Cross-linker reagents ontology","Fungal gross anatomy","FlyBase Controlled Vocabulary","Physico-chemical methods and properties","Flora Phenotype Ontology","Influenza ontology","Foundational Model of Anatomy Ontology (subset)","Food Ontology","Geographical Entity Ontology","Human Ancestry Ontology","Hymenoptera Anatomy Ontology","human phenotype ontology","Human Developmental Stages","Informed Consent Ontology","Infectious Disease Ontology","Livestock Breed Ontology","Mathematical Modelling Ontology","Minimum Information for A Phylogenetic Analysis (MIAPA) Ontology","Non-Coding RNA Ontology","nuclear magnetic resonance CV","The Prescription of Drugs Ontology","Platynereis Developmental Stages","PRIDE Controlled Vocabulary","Sustainable Development Goals Interface Ontology","Semanticscience Integrated Ontology","STATO: the statistical methods ontology","Symptom Ontology","Taxonomic rank vocabulary","Terminology for Description of Dynamics","Mosquito gross anatomy","Plant Trait Ontology","Teleost taxonomy ontology","Units of measurement ontology","Combined phenotype ontology","Variation Ontology","C. elegans phenotype","Experimental condition ontology","Zebrafish Experimental Conditions Ontology"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Prefix<\/th>\n      <th>Namespace<\/th>\n      <th>Title<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="a-brief-rols-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#a-brief-rols-overview" class="anchor"></a>A Brief rols overview</h1>
<p>The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package is build around a few classes that enable to query the OLS and retrieve, store and manipulate data. Each of these classes are described in more details in their respective manual pages. We start by loading the package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"rols"</span>)</code></pre></div>
<div id="ontologies" class="section level2">
<h2 class="hasAnchor">
<a href="#ontologies" class="anchor"></a>Ontologies</h2>
<p>The <code>Ontology</code> and <code>Ontologies</code> classes can store information about single of multiple ontologies. The latter can be easily subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ol &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontologies</a></span>()
ol</code></pre></div>
<pre><code>## Object of class 'Ontologies' with 214 entries
##    AERO, BFO ... XCO, ZECO</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/Ontology-class.html">olsNamespace</a></span>(ol))</code></pre></div>
<pre><code>##     aero      bfo   co_330   co_334   co_335   co_360 
##   "aero"    "bfo" "co_330" "co_334" "co_335" "co_360"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ol[[<span class="st">"go"</span>]]</code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2018-10-05 Updated: 2018-10-05 Version: 2018-10-03 
##    49709 terms  60 properties  0 individuals</code></pre>
<p>It is also possible to initialise a single ontology</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">go &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontology</a></span>(<span class="st">"go"</span>)
go</code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2018-10-05 Updated: 2018-10-05 Version: 2018-10-03 
##    49709 terms  60 properties  0 individuals</code></pre>
</div>
<div id="terms" class="section level2">
<h2 class="hasAnchor">
<a href="#terms" class="anchor"></a>Terms</h2>
<p>Single ontology terms are stored in <code>Term</code> objects. When more terms need to be manipulated, they are stored as <code>Terms</code> objects. It is easy to obtain all terms of an ontology of interest, and the resulting <code>Terms</code> object can be subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">terms</a></span>(go) ## or terms("go")
gotrms</code></pre></div>
<pre><code>## Object of class 'Terms' with 49710 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0032957, GO:0032942</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</code></pre></div>
<pre><code>## Object of class 'Terms' with 10 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0001819, GO:0044831</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms[[<span class="st">"GO:0090575"</span>]]</code></pre></div>
<pre><code>## A Term from the GO ontology: GO:0090575 
##  Label: RNA polymerase II transcription factor complex
##   A transcription factor complex that acts at a regulatory region of
##   a gene transcribed by RNA polymerase II.</code></pre>
<p>It is also possible to initialise a single term</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(go, <span class="st">"GO:0090575"</span>)
<span class="kw"><a href="../reference/Term-class.html">termId</a></span>(trm)</code></pre></div>
<pre><code>## [1] "GO:0090575"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(trm)</code></pre></div>
<pre><code>## [1] "RNA polymerase II transcription factor complex"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">strwrap</span>(<span class="kw"><a href="../reference/Term-class.html">termDesc</a></span>(trm))</code></pre></div>
<pre><code>## [1] "A transcription factor complex that acts at a regulatory region of"
## [2] "a gene transcribed by RNA polymerase II."</code></pre>
<p>It is then possible to extract the <code>ancestors</code>, <code>descendants</code>, <code>parents</code> and <code>children</code> terms. Each of these functions return a <code>Terms</code> object</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">parents</a></span>(trm)</code></pre></div>
<pre><code>## Object of class 'Terms' with 1 entries
##  From the GO ontology
## GO:0044798</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">children</a></span>(trm)</code></pre></div>
<pre><code>## Object of class 'Terms' with 41 entries
##  From the GO ontology
##   GO:0062071, GO:0008230 ... GO:0034718, GO:0030232</code></pre>
<p>Similarly, the <code>partOf</code> and <code>derivesFrom</code> functions return, for an input term, the terms it is a part of and derived from.</p>
<p>Finally, a single term or terms object can be coerced to a <code>data.frame</code> using <code>as(x, "data.frame")</code>.</p>
</div>
<div id="properties" class="section level2">
<h2 class="hasAnchor">
<a href="#properties" class="anchor"></a>Properties</h2>
<p>Properties (relationships) of single or multiple terms or complete ontologies can be queries with the <code>properties</code> method, as briefly illustrated below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(<span class="st">"uberon"</span>, <span class="st">"UBERON:0002107"</span>)
trm</code></pre></div>
<pre><code>## A Term from the UBERON ontology: UBERON:0002107 
##  Label: liver
##   An exocrine gland which secretes bile and functions in metabolism
##   of protein and carbohydrate and fat, synthesizes substances
##   involved in the clotting of the blood, synthesizes vitamin A,
##   detoxifies poisonous substances, stores glycogen, and breaks down
##   worn-out erythrocytes[GO].</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Properties-class.html">properties</a></span>(trm)
p</code></pre></div>
<pre><code>## Object of class 'Properties' with 112 entries
##  From the UBERON ontology
##   digestive system gland, abdomen element ... liver bud, hepatobiliary system</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p[[<span class="dv">1</span>]]</code></pre></div>
<pre><code>## A Property from the UBERON ontology: UBERON:0006925 
##  Label: digestive system gland</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(p[[<span class="dv">1</span>]])</code></pre></div>
<pre><code>## [1] "digestive system gland"</code></pre>
</div>
</div>
<div id="use-case" class="section level1">
<h1 class="hasAnchor">
<a href="#use-case" class="anchor"></a>Use case</h1>
<p>A researcher might be interested in the trans-Golgi network. Searching the OLS is assured by the <code>OlsSearch</code> and <code>olsSearch</code> classes/functions. The first step is to defined the search query with <code>OlsSearch</code>, as shown below. This creates an search object of class <code>OlsSearch</code> that stores the query and its parameters. In records the number of requested results (default is 20) and the total number of possible results (there are 15166 results across all ontologies, in this case). At this stage, the results have not yet been downloaded, as shown by the 0 responses.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 15166)
##   response(s): 0</code></pre>
<p>15166 results are probably too many to be relevant. Below we show how to perform an exact search by setting <code>exact = TRUE</code>, and limiting the search the the GO ontology by specifying <code>ontology = "GO"</code>, or doing both.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 8)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 755)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 1)
##   response(s): 0</code></pre>
<p>One case set the <code>rows</code> argument to set the number of desired results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">rows =</span> <span class="dv">200</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 200 (out of 755)
##   response(s): 0</code></pre>
<p>Alternatively, one can call the <code>allRows</code> function to request all results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 755)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">allRows</a></span>(tgnq))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 755 (out of 755)
##   response(s): 0</code></pre>
<p>Letâ€™s proceed with the exact search and retrieve the results. Even if we request the default 20 results, only the 8 relevant result will be retrieved. The <code>olsSearch</code> function updates the previously created object (called <code>qry</code> below) by adding the results to it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)
(qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">olsSearch</a></span>(qry))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 8)
##   response(s): 8</code></pre>
<p>We can now transform this search result object into a fully fledged <code>Terms</code> object or a <code>data.frame</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(qtrms &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"Terms"</span>))</code></pre></div>
<pre><code>## Object of class 'Terms' with 8 entries
##  From  8 ontologies
##   NCIT:C33802, OMIT:0020822 ... GO:0005802, GO:0005802</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(qdrf &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"data.frame"</span>))</code></pre></div>
<pre><code>## 'data.frame':    8 obs. of  10 variables:
##  $ id                  : chr  "ncit:class:http://purl.obolibrary.org/obo/NCIT_C33802" "omit:class:http://purl.obolibrary.org/obo/OMIT_0020822" "go:class:http://purl.obolibrary.org/obo/GO_0005802" "cco:http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ iri                 : chr  "http://purl.obolibrary.org/obo/NCIT_C33802" "http://purl.obolibrary.org/obo/OMIT_0020822" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ short_form          : chr  "NCIT_C33802" "OMIT_0020822" "GO_0005802" "GO_0005802" ...
##  $ obo_id              : chr  "NCIT:C33802" "OMIT:0020822" "GO:0005802" "GO:0005802" ...
##  $ label               : chr  "Trans-Golgi Network" "trans-Golgi Network" "trans-Golgi network" "trans-Golgi network" ...
##  $ description         :List of 8
##   ..$ : chr "A network of membrane components where vesicles bud off the Golgi apparatus to bring proteins, membranes and ot"| __truncated__
##   ..$ : NULL
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##  $ ontology_name       : chr  "ncit" "omit" "go" "cco" ...
##  $ ontology_prefix     : chr  "NCIT" "OMIT" "GO" "CCO" ...
##  $ type                : chr  "class" "class" "class" "class" ...
##  $ is_defining_ontology: logi  TRUE TRUE TRUE FALSE FALSE FALSE ...</code></pre>
<p>In this case, we can see that we actually retrieve the same term used across different ontologies. In such cases, it might be useful to keep only non-redundant term instances. Here, this would have been equivalent to searching the ncit, omit, go ontology</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qtrms &lt;-<span class="st"> </span><span class="kw">unique</span>(qtrms)
<span class="kw"><a href="../reference/Term-class.html">termOntology</a></span>(qtrms)</code></pre></div>
<pre><code>##  NCIT:C33802 OMIT:0020822   GO:0005802 
##       "ncit"       "omit"         "go"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termNamespace</a></span>(qtrms)</code></pre></div>
<pre><code>## $`NCIT:C33802`
## NULL
## 
## $`OMIT:0020822`
## NULL
## 
## $`GO:0005802`
## [1] "cellular_component"</code></pre>
<p>Below, we execute the same query using the <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"GO.db"</span>)
GOTERM[[<span class="st">"GO:0005802"</span>]]</code></pre></div>
<pre><code>## GOID: GO:0005802
## Term: trans-Golgi network
## Ontology: CC
## Definition: The network of interconnected tubular and cisternal
##     structures located within the Golgi apparatus on the side
##     distal to the endoplasmic reticulum, from which secretory
##     vesicles emerge. The trans-Golgi network is important in the
##     later stages of protein secretion where it is thought to play
##     a key role in the sorting and targeting of secreted proteins
##     to the correct destination.
## Synonym: Golgi trans face
## Synonym: Golgi trans-face
## Synonym: late Golgi
## Synonym: maturing face
## Synonym: TGN
## Synonym: trans face
## Synonym: trans Golgi network</code></pre>
</div>
<div id="on-line-vs-off-line-data" class="section level1">
<h1 class="hasAnchor">
<a href="#on-line-vs-off-line-data" class="anchor"></a>On-line vs.Â off-line data</h1>
<p>It is possible to observe different results with <em><a href="http://bioconductor.org/packages/rols">rols</a></em> and <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em>, as a result of the different ways they access the data. <em><a href="http://bioconductor.org/packages/rols">rols</a></em> or <em><a href="http://bioconductor.org/packages/biomaRt">biomaRt</a></em> perform direct online queries, while <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> and other annotation packages use database snapshot that are updated every release.</p>
<p>Both approaches have advantages. While online queries allow to obtain the latest up-to-date information, such approaches rely on network availability and quality. If reproducibility is a major issue, the version of the database to be queried can easily be controlled with off-line approaches. In the case of <em><a href="http://bioconductor.org/packages/rols">rols</a></em>, although the load date of a specific ontology can be queried with <code>olsVersion</code>, it is not possible to query a specific version of an ontology.</p>
</div>
<div id="changes-in-version-2-0" class="section level1">
<h1 class="hasAnchor">
<a href="#changes-in-version-2-0" class="anchor"></a>Changes in version 2.0</h1>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> 2.0 has substantially changed. While the table below shows some correspondence between the old and new interface, this is not always the case. The new interface relies on the <code>Ontology</code>/<code>Ontologies</code>, <code>Term</code>/<code>Terms</code> and <code>OlsSearch</code> classes, that need to be instantiated and can then be queried, as described above.</p>
<table class="table">
<thead><tr class="header">
<th>version &lt; 1.99</th>
<th>version &gt;= 1.99</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>ontologyLoadDate</code></td>
<td>
<code>olsLoaded</code> and <code>olsUpdated</code>
</td>
</tr>
<tr class="even">
<td><code>ontologyNames</code></td>
<td><code>Ontologies</code></td>
</tr>
<tr class="odd">
<td><code>olsVersion</code></td>
<td><code>olsVersion</code></td>
</tr>
<tr class="even">
<td><code>allIds</code></td>
<td><code>terms</code></td>
</tr>
<tr class="odd">
<td><code>isIdObsolete</code></td>
<td><code>isObsolete</code></td>
</tr>
<tr class="even">
<td><code>rootId</code></td>
<td><code>olsRoot</code></td>
</tr>
<tr class="odd">
<td><code>olsQuery</code></td>
<td>
<code>OlsSearch</code> and <code>olsSearch</code>
</td>
</tr>
</tbody>
</table>
<p>Not all functionality is currently available. If there is anything that you need but not available in the new version, please contact the maintained by opening an <a href="https://github.com/lgatto/rols/issues">issue</a> on the package development site.</p>
</div>
<div id="cvparams" class="section level1">
<h1 class="hasAnchor">
<a href="#cvparams" class="anchor"></a>CVParams</h1>
<p>The <code>CVParam</code> class is used to handle controlled vocabulary. It can be used for user-defined parameters</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">name =</span> <span class="st">"A user param"</span>, <span class="dt">value =</span> <span class="st">"the value"</span>)</code></pre></div>
<pre><code>## [, , A user param, the value]</code></pre>
<p>or official controlled vocabulary (which triggers a query to the OLS service)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">accession =</span> <span class="st">"MS:1000073"</span>)</code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"electrospray ionization"</span>)</code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"ESI"</span>) ## using a synonym</code></pre></div>
<pre><code>## [MS, MS:1000073, ESI, ]</code></pre>
<p>See <code><a href="../reference/CVParam-class.html">?CVParam</a></code> for more details and examples.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R version 3.5.1 Patched (2018-08-16 r75161)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.0
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] DT_0.4               rols_2.9.2           GO.db_3.6.0         
## [4] AnnotationDbi_1.42.1 IRanges_2.14.11      S4Vectors_0.18.3    
## [7] Biobase_2.40.0       BiocGenerics_0.26.0  BiocStyle_2.8.2     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.18      later_0.7.4       compiler_3.5.1   
##  [4] prettyunits_1.0.2 progress_1.2.0    tools_3.5.1      
##  [7] digest_0.6.17     bit_1.1-14        jsonlite_1.5     
## [10] evaluate_0.11     memoise_1.1.0     RSQLite_2.1.1    
## [13] pkgconfig_2.0.2   rlang_0.2.2       shiny_1.1.0      
## [16] DBI_1.0.0         crosstalk_1.0.0   commonmark_1.5   
## [19] curl_3.2          yaml_2.2.0        pkgdown_1.1.0    
## [22] xfun_0.3          httr_1.3.1        stringr_1.3.1    
## [25] roxygen2_6.1.0    xml2_1.2.0        knitr_1.20       
## [28] htmlwidgets_1.2   hms_0.4.2         desc_1.2.0       
## [31] fs_1.2.6          rprojroot_1.3-2   bit64_0.9-7      
## [34] R6_2.2.2          rmarkdown_1.10    bookdown_0.7     
## [37] blob_1.1.1        magrittr_1.5      promises_1.0.1   
## [40] backports_1.1.2   htmltools_0.3.6   MASS_7.3-50      
## [43] assertthat_0.2.0  xtable_1.8-3      mime_0.5         
## [46] httpuv_1.4.5      stringi_1.2.4     crayon_1.3.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#getting-help">Getting help</a></li>
      <li><a href="#the-resource">The resource</a></li>
      <li><a href="#the-package">The package</a></li>
      </ul>
</li>
      <li>
<a href="#a-brief-rols-overview">A Brief rols overview</a><ul class="nav nav-pills nav-stacked">
<li><a href="#ontologies">Ontologies</a></li>
      <li><a href="#terms">Terms</a></li>
      <li><a href="#properties">Properties</a></li>
      </ul>
</li>
      <li><a href="#use-case">Use case</a></li>
      <li><a href="#on-line-vs-off-line-data">On-line vs.Â off-line data</a></li>
      <li><a href="#changes-in-version-2-0">Changes in version 2.0</a></li>
      <li><a href="#cvparams">CVParams</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
